AppCenterCrashes/AppCenterCrashesTests/MSCrashesCategoryTests.m

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "MSAppCenterInternal.h"
+#import "MSCrashesCategory.h"
+#import "MSCrashesPrivate.h"
+#import "MSMockUserDefaults.h"
+#import "MSTestFrameworks.h"
+#import "MSUtility+Application.h"
+
+#if TARGET_OS_OSX
+
+@interface NSApplication (CrashException)
+
+- (void)ms_reportException:(NSException *)exception;
+- (void)ms_sendEvent:(NSEvent *)theEvent;
+
+@end
+
+#endif
+
+static NSException *lastException;
+static void exceptionHandler(NSException *exception) {
+  lastException = exception;
+}
+
+@interface MSCrashesCategoryTests : XCTestCase
+
+@end
+
+@implementation MSCrashesCategoryTests
+
+- (void)testActivateCategory {
+
+  // If
+#if TARGET_OS_OSX
+  MSMockUserDefaults *settings = [MSMockUserDefaults new];
+  [settings setObject:@YES forKey:@"NSApplicationCrashOnExceptions"];
+  id applicationMock = OCMPartialMock([NSApplication sharedApplication]);
+  id appCenterMock = OCMClassMock([MSAppCenter class]);
+  OCMStub([appCenterMock isDebuggerAttached]).andReturn(NO);
+  id crashesMock = OCMPartialMock([MSCrashes sharedInstance]);
+  OCMStub([crashesMock exceptionHandler]).andReturn((NSUncaughtExceptionHandler *)exceptionHandler);
+#endif
+  // When
+  [MSCrashesCategory activateCategory];
```

Comment:
This line gets zoned out by `#if`s in case it's not `TARGET_OS_OSX`. Is that intended?

Reply:
yes